### PR TITLE
feat(shorebird_cli): Use diffs for iOS patching

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

Adds support for diff patching on iOS.

Depends on:

- https://github.com/shorebirdtech/updater/pull/111
- https://github.com/shorebirdtech/engine/pull/72

Part of https://github.com/shorebirdtech/shorebird/issues/675

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_cli/lib/src/commands/patch/patch_ios_framework_command.dart [hunk 1]
```diff
@@ -194,18 +211,73 @@ Please re-run the release command for this version or create a new release.''');
       return ExitCode.software.code;
     }
 
+    // TODO use linker if available
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/commands/patch/patch_ios_framework_command.dart:214]: Does this break linking?

File: packages/shorebird_cli/lib/src/executables/aot_tools.dart [hunk 2]
```diff
@@ -52,4 +58,56 @@ class AotTools {
...
+  /// `--dump-blobs` flag.
+  Future<bool> isGeneratePatchDiffBaseSupported() async {
+    // This will always return a non-zero exit code because the input is a
+    // (presumably) nonexistent file. If the --dump-blobs flag is supported,
+    // the error message will contain something like: "Snapshot file does not
+    // exist". If the flag is not supported, the error message will contain
+    // "Unrecognized flags: dump_blobs"
+    final result = await _exec(
+      [
+        'dump_blobs',
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/executables/aot_tools.dart:72]: Do you need any of the args at all?

**Review by eseidel — APPROVED** [review]

File: packages/shorebird_cli/lib/src/commands/patch/patch_ios_framework_command.dart (hunk 1 above)
Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/commands/patch/patch_ios_framework_command.dart:214]: To clarify, the question is "does differential patching break linking for all iOS builds"? I believe the answer is no (we successfully link and apply vmcode patches in the patch_ios_command)

File: packages/shorebird_cli/lib/src/executables/aot_tools.dart (hunk 2 above)
Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/executables/aot_tools.dart:72]: Yep:

```
bryanoltman@boltman ~/shorebirdtech/dart-sdk/sdk (23083f76dd7)
⑆ ./xcodebuild/DebugARM64/analyze_snapshot --dump_blobs
At least one input is required
Usage: analyze_snapshot [<vm-flags>] [<options>] <snapshot_data>

Common options:
--help
  Display this message.
--sdk_version
  Print the SDK version.
--out
  Path to generate the analysis results JSON.
--shorebird
  Produce Shorebird formatted JSON, requires --out.
If omitting [<vm-flags>] the VM parsing the snapshot is created with the
following default flags:
--enable_mirrors=false
--background_compilation
--precompilation



bryanoltman@boltman ~/shorebirdtech/dart-sdk/sdk (23083f76dd7)
⑆ ./xcodebuild/DebugARM64/analyze_snapshot --dump_blobs out
No output path provided.

bryanoltman@boltman ~/shorebirdtech/dart-sdk/sdk (23083f76dd7)
⑆ ./xcodebuild/DebugARM64/analyze_snapshot --dump_blobs --out=asdf.json out
Setting VM flags failed: Unrecognized flags: dump_blobs
```

File: packages/shorebird_cli/lib/src/commands/patch/patch_ios_framework_command.dart (hunk 1 above)
Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/commands/patch/patch_ios_framework_command.dart:214]: Resolved offline, this just means this command hasn't implemented support for linking with aot-tools yet. Filed https://github.com/shorebirdtech/shorebird/issues/1640 to track

**Review by eseidel — APPROVED** [review]

**Pull request merged**